[RNMobile] Add Details Panel to VideoPress Block Settings

SiobhyB opened the pull request:
### Related PRs:

* Relies on changes in this Gutenberg PR: https://github.com/WordPress/gutenberg/pull/48737

### Proposed changes:

* A [native version of VideoPress' details panel](https://github.com/Automattic/jetpack/blob/e74058d26104f8804201babec7e3a7477f81e4fb/projects/packages/videopress/src/client/block-editor/blocks/video/components/details-panel/index.native.tsx) has been created and included in the block's settings. The native version includes options to edit the video's title and description.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

Designs for block settings can be found here: p9ugOq-3tP-p2

## Does this pull request change what data or activity we track or use?

No, it doesn't.

## Testing instructions:

### Empty state

* With these branches checked out, open the editor and add a new VideoPress block in either app.
* Tap the settings toggle and verify that the details panel displays, with title and description fields that open up an editable subsheet when tapped.
* Add a new title and description to the block, save, and confirm that they remain in place.

### Existing title and description

* Go through the steps to add the VideoPress block on the web, and make sure to add a title and description.
* Open up the app and navigate to the block that added via the web.
* Tap the settings toggle to verify that the block's title and description are visible.
* Edit the title and description then save, confirm that the updated values appear as expected in both the app and the web.

### Screenshots: Empty state 

| Main View  | Subsheet |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/222786729-81497400-7693-4ddf-92a8-931be99b7469.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/222786770-26a54c78-837e-4f0f-8cf3-a33c987c135c.png" width="100%"> |

### Screenshots: Existing title and description 

| Main View  | Subsheet |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/222787439-a2b4c8ce-eb9b-4f98-a554-de6037a9742f.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/222787489-b90dfe2a-e8ab-4735-8598-47678a4cfd1c.png" width="100%"> |

SiobhyB committed:
Add settings toggle to VideoPress block

With this commit, the settings toggle is added to the block when it's selected. The settings will be iterated on in future PRs.

SiobhyB committed:
Add changelog entry

SiobhyB committed:
Merge branch 'trunk' into rnmobile/add-videopress-block-settings-toggle

SiobhyB committed:
Update changelog to reflect change's significance

SiobhyB committed:
Add DetailsPanel component to block's settings

SiobhyB committed:
Implement native-specific DetailsPanel component

With this commit, a native index file is added to the DetailsPanel component. The original code in the default index file has been modified to include components that work with React Native.

SiobhyB committed:
Set a different placeholder for component's cell

fluiddot commented on projects/packages/videopress/src/client/block-editor/blocks/video/components/details-panel/index.native.tsx:
```diff
@@ -0,0 +1,42 @@
+
+	return (
+		<PanelBody title={ __( 'Details', 'jetpack-videopress-pkg' ) }>
+			<BottomSheetTextControl
```
I think this component is generally used for multiline content, however, in this case, the title seems a one-line value. I'm wondering if we could use a different component like [`TextControl`](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/text-control/README.md), WDYT?

SiobhyB replied:
Good point, that'd make sense to me! 

@osullivanchris, 👋 , I just want to check in with you before making this change. Does the one-line component for the title field make sense to you, too? I've created a video of what it'd look like:

https://user-images.githubusercontent.com/2998162/224992576-4e19399c-ba59-4f8e-a415-86f99137c002.mov

I also slightly tweaked the placeholder text in this case, as having the same text in the new components made the alignment feel a bit off to me: 

| `None` placeholder  | `Add..` placeholder |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/224993096-33d02749-628b-4ac3-a30f-afe3084e71db.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/224992921-9a7871cf-25cc-4788-a8d4-8a99df74eb40.png" width="100%"> |

Happy for you to have the final call on this @osullivanchris, thanks! 🙇‍♀️

osullivanchris replied:
@SiobhyB @fluiddot hey this seems like a good idea to me. Happy with that change. Noting that it might mean some of the title is not visible if its long. But the user can still select within it to see the other characters. So it seems worth the benefit of not having to drill down 👍

SiobhyB replied:
Thanks both! I've gone ahead to make that change in https://github.com/Automattic/jetpack/pull/29283/commits/80804d4940839637381361085297efd408d21370 and have re-requested your review @fluiddot :)

fluiddot replied:
Great, thanks @SiobhyB for making the change 🙇. I've just reviewed and approved the PR :shipit: .

fluiddot reviewed (commented):
@SiobhyB I tested the changes and look great. I only added a comment regarding how to display the `title` attribute. Thanks!

SiobhyB committed:
refactor: multi-line → single-line input for title

This commit follows the feedback provided here: https://github.com/Automattic/jetpack/pull/29283#discussion_r1129868456

The title field previously used a multi-line input field, this commit replaces that field with a single-line input.

SiobhyB committed:
Merge branch 'trunk' into rnmobile/videopress-block-add-details-settings

SiobhyB committed:
Update and rename changelog file

The previous changelog was inherited from the old parent branch, but doesn't match this PR's changes. It has therefore been updated for accuracy.

fluiddot reviewed (approved):
LGTM 🎊 !

I followed the testing instructions and the changes worked as expected ✅ .

nunyvega commented on projects/packages/videopress/src/client/block-editor/blocks/video/components/details-panel/index.native.tsx:
```diff
@@ -0,0 +1,40 @@
+				label={ __( 'Title', 'jetpack-videopress-pkg' ) }
+			/>
+			<BottomSheetTextControl
+				initialValue={ description }
```
If the description is not set, it has no fallback value, should we add one?

SiobhyB replied:
@nunyvega, thanks for taking a look at this! There is a fallback in the `BottomSheetTextControl` component here, so I don't think it's necessary to add that to `initialValue` here: 

https://github.com/WordPress/gutenberg/blob/43338de43993d286879012021b6857a700cc2e40/packages/components/src/mobile/bottom-sheet-text-control/index.native.js#L63

nunyvega replied:
Makes sense! marking as resolved

nunyvega reviewed (approved):
LGTM
🚢
